### PR TITLE
Refactored recover.weaker.R

### LINKED ISSUE #74: refactor recover.weaker.R

### PR DESCRIPTION
**Pull request by hechth**
- Extracted all functions performing substeps of the computation
- Renamed variables to better reflect the data they actually contain
- Disabled parallelism in test case for better debugging
- Working toward isolating executions on individual files to ease implementing a parallel version in Galaxy
- Added documentation to all extracted functions
- Added stubs for further simplifications which might be more drastical code changes --> those need to be validated in extended tests

Closes #74

### CODE REVIEW

File: R/recover.weaker.R
Comment (maximskorik) [inline R/recover.weaker.R:32]: Maybe it would be better to use a `for` loop, since that's what it essentially does.

File: R/recover.weaker.R
Comment (maximskorik) [inline R/recover.weaker.R:35]: Instead of assigning `1`s to the indices that have to be removed, `to.remove` can just store these indices. This line would then look simply like this:
```R
new.table <- new.table[-to.remove, ]
```

File: R/recover.weaker.R
Comment (maximskorik) [inline R/recover.weaker.R:432]: This pattern seems to be occurring several times. Maybe we could extract it to a function that returns a vector `c(lower, upper)`, akin to R's `range()`?

Comment (maximskorik) [reply R/recover.weaker.R:432]: After having another look I realised I was probably wrong. In the other case it computed indices, not values.

**Review by maximskorik — APPROVED** [review]
Looks good! I left a couple of comments where I think the code can be improved. Can be merged after the CI passes.

File: R/recover.weaker.R
Comment (hechth) [reply R/recover.weaker.R:32]: Absolutely!

File: R/recover.weaker.R
Comment (zargham-ahmad) [inline R/recover.weaker.R:676]: from performance perspective I think we can also explore using data.table with Indexes

File: R/adaptive.bin.R
Comment (zargham-ahmad) [inline R/adaptive.bin.R:140]: I think idk maybe we should try to find a balance between using dplyr function and base function especially when using base function doesn't make code ugly and less readable because dplyr function are quite slow.

File: R/recover.weaker.R
Comment (zargham-ahmad) [inline R/recover.weaker.R:43]: #' in the mid is it a typo?

**Review by zargham-ahmad — APPROVED** [review]
Looks good to me. Can be merged after the CI passes.

File: R/recover.weaker.R
Comment (hechth) [reply R/recover.weaker.R:43]: No it is a comment on A new line -if you view this on a mobile device it just looks weird. Or what do you mean specifically?

**Comment by hechth** [comment]
The CI still fails due to problems with the hybrid test - I'll still address these issues before we can merge this. The problem seems to be related to inconsistent use of data frames, lists and tibbles due to the introduced refactorings in different places.

File: R/recover.weaker.R
Comment (zargham-ahmad) [reply R/recover.weaker.R:43]: I am taking about the one in the middle on line 43 `between#' neighbouring`, is it for the comment on the new line?

File: R/recover.weaker.R
Comment (hechth) [reply R/recover.weaker.R:676]: with indexes you mean numbers and not column names? I'd be highly against that, unless we define for example global variables that define the indices with names

File: R/recover.weaker.R
Comment (hechth) [reply R/recover.weaker.R:43]: Yes that one is!

**Pull request merged**
